src/graph/graph.c

```diff
@@ -25,20 +26,31 @@ void _Graph_LeaveCriticalSection(Graph *g) {
     pthread_mutex_unlock(&g->_mutex);
 }
 
+/* Force execution of all pending operations on a matrix. */
+static inline void _Graph_SynchronizeMatrix(GrB_Matrix m) {
+    GrB_Index nvals;
+    GrB_Matrix_nvals(&nvals, m);
+}
+
 /* Resize given matrix, such that its number of row and columns
- * matches the number of nodes in the graph. */
+ * matches the number of nodes in the graph. Also, synchronize
+ * matrix to execute any pending operations. */
 void _Graph_ResizeMatrix(const Graph *g, GrB_Matrix m) {
```

Comment:
Now that this function performs resize and commits pending changes I think we need to rename it.